File: controllers/obs_remote.js
const OBSWebSocket = require('obs-websocket-js');

//Setting obs-websocket up
const obs = new OBSWebSocket();
const remote = require('../obs/obs_remote');

exports.remote_get_open_connection = (req, res, next) => {
    remote.openConnection(obs)
        .then(result => {
            res.status(200).json({
                message: "Connection to obs succeeded!",
                data: result
            });
        })
        .catch(err => {
            res.status(500).json({
                error: "Couldn't connect to obs!"
            });
        });
};

exports.remote_get_close_connection = (req, res, next) => {
    if(remote.closeConnection(obs)){
        res.status(200).json({
            message: "Connection to obs gone down!"
        });
    } else {
        res.status(500).json({
            error: "Closing connection to obs failed!"
        });
    }
}

exports.remote_post_start_recording = (req, res, next) => {
    remote.startRecording(obs)
        .then(result => {
            res.status(200).json({
                message: "You're now recording! \n " +
                    "Recording started at : " + result.startedAt,
                data: result
            });
        })
        .catch(err => {
            res.status(500).json({
                message: "Starting recording failed!",
                error: err
            });
        });
};

exports.remote_post_stop_recording = (req, res, next) => {
    remote.stopRecording(obs)
        .then(result => {
            res.status(200).json({
                message: "Recording has been stopped! " +
                    "It started at " + remote.infos.startedAt +
                    "Record duration : " + remote.infos.recorded.toString(),
                data: result
            });
        })
        .catch(err => {
            res.status(500).json({
                message: "Stoping recording failed!",
                error: err
            });
        });
};

exports.remote_get_scenes = (req, res, next) => {
    remote.getScenes(obs)
        .then(result => {
            res.status(200).json({
                data: result
            });
        })
        .catch(err => {
            res.status(500).json({
                message: "An error occured",
                error: err
            });
        });
};

exports.remote_get_current_scene = (req, res, next) => {
    remote.getCurrentScene(obs)
        .then(result => {
            res.status(200).json({
                data: result
            });
        })
        .catch(err => {
            res.status(500).json({
                message: "An error occured",
                error: err
            });
        });
};

exports.remote_post_set_current_scene = (req, res, next) => {
    console.log('API.Controller - setting current scene to ' + req.query['scene-name']);
    remote.setCurrentScene(obs, req.query['scene-name'])
        .then(result => {
            res.status(200).json({
                status: 'ok'
            });
            console.log("API - Scene have been changed");
        })
        .catch(err => {
            res.status(500).json({
                message: "An error occured",
                error: err
            });
        });
};

File: quarkus-api/src/main/java/zenika/obs/api/ObsRemote.java
package zenika.obs.api;

import java.io.Serializable;
import java.util.HashSet;
import java.util.Set;

public class ObsRemote implements Serializable {
    private Boolean connected;
    private Boolean recording;
    private String currentScene;
    private Set<String> scenes;

    public ObsRemote() {
        this.connected = false;
        this.recording = false;
        this.currentScene = "No current scene";
        this.scenes = new HashSet<>();
    }

    public Boolean isConnected() {
        return connected;
    }

    public void setConnected(Boolean connected) {
        this.connected = connected;
    }

    public Boolean isRecording() {
        return recording;
    }

    public void setRecording(Boolean recording) {
        this.recording = recording;
    }

    public String getCurrentScene() {
        return currentScene;
    }

    public void setCurrentScene(String currentScene) {
        this.currentScene = currentScene;
    }

    public Set<String> getScenes() {
        return scenes;
    }

    public void setScenes(Set<String> scenes) {
        this.scenes = scenes;
    }

    public void addScene(String scene) {
        this.scenes.add(scene);
    }

    @Override
    public String toString() {
        return "{" +
                "connected:" + connected +
                ", recording:" + recording +
                ", currentScene:'" + currentScene + '\'' +
                ", scenes:" + scenes +
                '}';
    }
}
